engine.py

```diff
@@ -504,9 +504,53 @@ def save_to_path(self, document, path):
 
             # make the document being processed the active document
             self.adobe.app.activeDocument = document
-
-            # TODO need to use appropriate save options here (jpg, tiff, etc)
-            document.saveAs(self.adobe.File(path))
+        
+            (_, ext) = os.path.splitext(path)
+
+            # valid file extensions and respective save option in photoshop 
+            valid_extensions = {
+                                 ".bmp" : self.adobe.BMPSaveOptions, 
+                                 # DCS1_saveOptions and DCS2_SaveOptions skipped
+                                 ".eps" : self.adobe.EPSSaveOptions,
+                                 ".gif" : self.adobe.GIFSaveOptions,
+
+                                 ".jpg" : self.adobe.JPEGSaveOptions,
+                                 ".jpeg" : self.adobe.JPEGSaveOptions,
+
+                                 ".pdf" : self.adobe.PDFSaveOptions,
+                                 ".psd" : self.adobe.PhotoshopSaveOptions,
+
+                                 ".pict" : self.adobe.PICTFileSaveOptions,
+                                 ".pct" : self.adobe.PICTFileSaveOptions,
+                                 ".pic" : self.adobe.PICTFileSaveOptions,
+                                 # PICTResourceSaveOptions skipped
+
+                                 ".pixar" : self.adobe.PixarSaveOptions,
```

Comment:
Looks like some of these are grouped by save options. I suggest either alphabetizing by the extension with no spaces.